gammapy/datasets/map.py

```diff
@@ -1360,6 +1311,79 @@ def reset_data_cache(self):
             if self.__dict__.pop(name, False):
                 log.info(f"Clearing {name} cache for dataset {self.name}")
 
+    def group_over_energy(self, e_edges=None, name=None):
+        """Group MapDataset over reco energy edges.
+
+        Counts are summed taking into account safe mask.
+
+        Parameters
+        ----------
+        e_edges : `~astropy.units.Quantity`
+            the energy edges.
+        name: str
+            Name of the new dataset.
+
+        Returns
+        -------
+        dataset: `MapDataset`
+            Grouped dataset .
+        """
+        if e_edges is None:
+            e_axis = self._geom.get_axis_by_name("energy")
+            e_edges = u.Quantity([e_axis.edges[0], e_axis.edges[-1]])
+
+        axis = MapAxis.from_edges(e_edges, name="energy")
```

Comment:
I know that we have the `e_edges` argument elsewhere in Gammapy, but by now I think it would be the better option to just directly take a `MapAxis` object, as the first thing we do in many places is to create a `MapAxis`...

Reply:
Indeed passing a `MapAxis` directly might be more relevant. 
Note that there might be some confusion with `downsample(axis=None)` which expects an axis name rather than an axis per-se.

Reply:
Yes, good point! Maybe we should re-name those argument to `axis_name` uniformly...maybe open a reminder issue?